src/network/ifaddrs_patch.cpp

```diff
@@ -0,0 +1,129 @@
+/*!
+* Copyright (c) 2009, Jay Loden, Dave Daeschler, Giampaolo Rodola.
+ * Licensed under the BSD 3-Clause License.
+ * See https://github.com/giampaolo/psutil/blob/master/LICENSE
+ */
+#if defined(sun) || defined(__sun)
```

Comment:
I believe that Solaris check is 
```
if (defined(sun) || defined(__sun)) && (defined(__SVR4) || defined(__svr4__))
```

The current check catches SunOS as well as Solaris which I think we don't want to patch here.

Actually, we doc't need to patch even Solaris 11, but this is not so trivial to find a version.

Reply:
yeah you're right, I can fix that

Reply:
fixed in https://github.com/microsoft/LightGBM/pull/3338/commits/465fbc4d982185eb74706fc6eba17ca341b1d7e4!

I tried testing on Solaris with R Hub, and all checks passed!

![image](https://user-images.githubusercontent.com/7608904/95292346-e8af7080-0836-11eb-9b85-f0e859b6a56e.png)